### ISSUE TITLE
License

### ISSUE THREAD

**Issue by TomasJohansson** [reneargento/algorithms-sedgewick-wayne#1]
Would you like to specify a license for the code in github repository "reneargento/algorithms-sedgewick-wayne" ?

I can mention that I am particularly interested in your implementation of "k shortest paths".
(i.e. your class "chapter4.section4.KShortestPaths" and its dependencies)

I have done some experimentation with it, comparing it (using the same test data) with the following three other libraries with implementations of "k shortest paths":
 https://github.com/yan-qi/k-shortest-paths-java-version
 https://github.com/bsmock/k-shortest-paths
 https://github.com/jgrapht/jgrapht	

Your implementation seem to be faster than those three above, so therefore I would be interested in using your implementation as a new Adapter implementation, i.e. a new implementation in this repository:
https://github.com/TomasJohansson/adapters-shortest-paths

However, the question is if you want to choose a license, to specify if/how others are allowed to use your code?

**Comment by reneargento** [comment]
Sure, I just added the MIT license to this repository.
I am happy to know about the performance of the KShortestPaths algorithm. If you ever publish your experiment results somewhere, I would be interested in reading them.

**Comment by TomasJohansson** [comment]
Thank you, I have now added an adapter using your implementation.

The biggest graph among my current test cases has 11815 vertices and 28320 edges.
(defined in a file "small_road_network_01.xml" https://github.com/TomasJohansson/adapters-shortest-paths/tree/c9ca982085d723eff2279f5f2a8367eb2bcabf8f/adapters-shortest-paths-test/src/test/resources/test_graphs/origin_bsmock )
When a test case search for the 10 shortest paths, all those shortest paths have between 123 - 131 edges.

Here are the results for the three fastest implementations when I ran them on my machine:
4 seconds: "ReneArgento"
8 seconds: "YanQi"
189 seconds: "Bsmock"
https://github.com/TomasJohansson/adapters-shortest-paths/blob/c9ca982085d723eff2279f5f2a8367eb2bcabf8f/adapters-shortest-paths-test/src/test/java/com/programmerare/shortestpaths/graph/tests/XmlDefinedTestCasesTest.java#L170
